src/client/pydaos/pydaos_shim.c

```diff
@@ -560,7 +560,11 @@ __shim_handle__kv_get(PyObject *self, PyObject *args)
 			} else if (evp->ev_error == -DER_REC2BIG) {
 				char *new_buff;
 
-				D_REALLOC(new_buff, op->buf, op->size);
+				/** No need to clear the buffer so size doesn't
+				 * matter.
+				 */
+				D_REALLOC(new_buff, op->buf, op->size,
+					  op->size);
```

Comment:
I'm not sure I like this, are you saying that as you don't want the buffer memset() then you can pass in the new size as the size?

As there's no need to clear the data here I think D_FREE/D_ALLOC would probably make more sense.

Reply:
Add a "zero" flag to

    D_REALLOC_COMMON(..., zero)

Add a set of special NOZERO APIs (for suitable use cases or performance):

    D_REALLOC_NOZERO(new_pointer, old_pointer, new_size)
    D_REALLOC_ARRAY_NOZERO(new_pointer, old_pointer, new_count)

Keep the default APIs' behavior as consistently zeroing new memory (for safety).

Reply:
@ashleypittman - I think the intention is to not clear the expanded part of the buffer. With D_FREE/D_ALLOC, you would lose information in the original buffer, right?

Having a specific api for not clearing the buffer, as liw suggests, is probably easier to understand and read.

Reply:
I think in this case the fetch failed as the buffer was too small, so it's going to retry with a bigger buffer.  There is no information in the original buffer.

Having a different API for NZ vs regular does make sense but I'm also mindful there are only ~20 calls to there macros so I'm not sure if it's worth the complexity, particularly if the ones that wouldn't use the NZ versions can be trivially replaced with DFREE/D_ALLLOC like I think this one could.

Reply:
The problem is D_ALLOC does a memset and isn't quite as efficient as realloc (which may essentially do nothing).   Perhaps we should have D_ALLOC_NZ if we go down this path.  There are certainly cases where calloc is pointless in our code.